command/state_mv.go

```diff
@@ -358,6 +355,17 @@ func (c *StateMvCommand) Run(args []string) int {
 	return 0
 }
 
+func eachModeForInstanceKey(key addrs.InstanceKey) states.EachMode {
+	switch key.(type) {
+	case addrs.IntKey:
+		return states.EachList
+	case addrs.StringKey:
+		return states.EachMap
+	default:
+		return states.NoEach
```

Comment:
Rather than defaulting to `NoEach` here, should we instead continue to treat that situation as an error so that in the unlikely event we add a new key type in future it won't just silently do something weird? Since it's a "should never happen" case in a codepath that doesn't run concurrently with other operations, a `panic` would probably be sufficient.

(If we do that then we would of course need a case for _no_ key to return `states.NoEach`, because that _is_ a valid situation.)